extensions/amp-social-share/1.0/social-share.js

```diff
@@ -24,33 +24,36 @@ import {getSocialConfig} from './social-share-config';
 import {openWindowDialog} from '../../../src/dom';
 import {useResourcesNotify} from '../../../src/preact/utils';
 
+const NAME = 'SocialShare';
 const DEFAULT_WIDTH = 60;
 const DEFAULT_HEIGHT = 44;
 const DEFAULT_TARGET = '_blank';
-const NAME = 'SocialShare';
+const WINDOW_FEATURES = 'resizable,scrollbars,width=640,height=480';
 
 /**
- * @param {!JsonObject} props
- *  type: !string,
- *  endpoint: ?string,
- *  params: ?JsonObject,
- *  target: ?string,
- *  width: ?string,
- *  height: ?string,
- *  color: ?string,
- *  background: ?string,
- *  tabIndex: ?string,
- *  style: ?string,
+ * @param {!SocialShareProps} props
  * @return {PreactDef.Renderable}
  */
-export function SocialShare(props) {
+export function SocialShare({
+  type,
+  endpoint,
+  params,
+  target,
+  width,
+  height,
+  color,
+  background,
+  tabIndex,
+  style,
+  children,
+}) {
   useResourcesNotify();
   const {
     finalEndpoint,
     checkedWidth,
     checkedHeight,
     checkedTarget,
-  } = checkProps(props);
+  } = checkProps(type, endpoint, target, width, height, params);
```

Comment:
You should still be able to pass `props` -- would be easier if you ever need to `checkProps` on anything new without having to change the fn signature.

Reply:
Note that `DateDisplay` is an example of a component that passes `props` to helper functions.

Reply:
Thanks!  Right we definitely can still use props.  One thing, I was thinking is that we do lose all the type definitions for these internal functions if we change to props.  I'm don't feel strongly about it either way, but I am thinking we can use this PR to clearly define all types within Social Share.

I also noticed that some of our bento components do not use this `prop` pattern such as `timeago`, `fit-text`.  Thoughts?

Reply:
Why would we lose the type definitions if we change to props? Won't they still be defined as properties of the prop definition provided?

Reply:
IMO `checkProps` in `SocialShare` is ostensibly more generic than `getFuzzyTimestampValue` whose inputs are unlikely to change. But this is up to you.

Reply:
Sounds good, since I just defined all the types for `checkProps` and `processChildren` functions in this PR, I'd prefer not to call them with `props` :)